client/src/game/star.js

```diff
@@ -397,8 +425,9 @@ class Star extends EventEmitter {
     }
 
     if (!this.text_garrison) {
-      let style = TextureService.DEFAULT_FONT_STYLE
-      style.fontSize = 4
+      let style = Object.assign({}, TextureService.DEFAULT_FONT_STYLE)
+      let small = this.zoomDepth > 2 || this.isSelected
```

Comment:
Create 2 instances of the garrison text instead, one large and one small. It may feel backwards but its better for performance as you will not have to re-draw the star.